### ISSUE TITLE
nix-build-perl-version-0.9912.drv-0 fails to build on unstable

### ISSUE THREAD

**Issue by sjau** [NixOS/nixpkgs#25229]
## Issue description
Perl 0.9912 seems to fail to build on current unstable:

```
patching ./vutil/lib/version/vxs.pm...

trying https://github.com/QuiteRSS/quiterss/archive/0.18.3.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
trying https://dl.google.com/android/repository/sources-19_r02.zip

trying http://download.skype.com/linux/skype-4.3.0.37.tar.bz2
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0Testing if you have a C compiler
Checking if your kit is complete...
Looks good
Generating a Unix-style Makefile
Writing Makefile for version::vxs
Writing MYMETA.yml and MYMETA.json
Generating a Unix-style Makefile
Writing Makefile for version
Writing MYMETA.yml and MYMETA.json
grep: Invalid range end
no configure script, doing nothing
building
make flags: SHELL=/nix/store/53h800j8kgpj0a349f7wxa5hgkj1vby2-bash-4.4-p12/bin/bash
cp lib/version.pod blib/lib/version.pod
cp lib/version.pm blib/lib/version.pm
cp vperl/vpp.pm blib/lib/version/vpp.pm
cp lib/version/regex.pm blib/lib/version/regex.pm
cp lib/version/Internals.pod blib/lib/version/Internals.pod
make[1]: Entering directory '/tmp/nix-build-perl-version-0.9912.drv-0/version-0.9912/vutil'
cp lib/version/vxs.pm ../blib/lib/version/vxs.pm
Running Mkbootstrap for version::vxs ()
chmod 644 "vxs.bs"
cc -c   -D_REENTRANT -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/no-such-path/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2   -DVERSION=\"0.9912\" -DXS_VERSION=\"0.9912\" -fPIC "-I/nix/store/xlw7glm144q9k26r3v351i31z2rp0b99-perl-5.24.1/lib/perl5/5.24.1/x86_64-linux-thread-multi/CORE"   vutil.c
100   125    0   125    0     0    119      0 --:--:--  0:00:01 --:--:--   119
"/nix/store/xlw7glm144q9k26r3v351i31z2rp0b99-perl-5.24.1/bin/perl" "/nix/store/xlw7glm144q9k26r3v351i31z2rp0b99-perl-5.24.1/lib/perl5/5.24.1/ExtUtils/xsubpp"  -typemap "/nix/store/xlw7glm144q9k26r3v351i31z2rp0b99-perl-5.24.1/lib/perl5/5.24.1/ExtUtils/typemap"  vxs.xs > vxs.xsc && mv vxs.xsc vxs.c
cc -c   -D_REENTRANT -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/no-such-path/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2   -DVERSION=\"0.9912\" -DXS_VERSION=\"0.9912\" -fPIC "-I/nix/store/xlw7glm144q9k26r3v351i31z2rp0b99-perl-5.24.1/lib/perl5/5.24.1/x86_64-linux-thread-multi/CORE"   vxs.c
rm -f ../blib/arch/auto/version/vxs/vxs.so
cc  -shared -O2 -L/nix/store/78w6a8zinhp7awdmrh5qbvxyci2pb8b4-glibc-2.25/lib -fstack-protector-strong vutil.o vxs.o  -o ../blib/arch/auto/version/vxs/vxs.so         \
        \

chmod 755 ../blib/arch/auto/version/vxs/vxs.so
"/nix/store/xlw7glm144q9k26r3v351i31z2rp0b99-perl-5.24.1/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- vxs.bs ../blib/arch/auto/version/vxs/vxs.bs 644
make[1]: Leaving directory '/tmp/nix-build-perl-version-0.9912.drv-0/version-0.9912/vutil'
  0 3253k    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0Manifying 2 pod documents
running tests
check flags: SHELL=/nix/store/53h800j8kgpj0a349f7wxa5hgkj1vby2-bash-4.4-p12/bin/bash
make[1]: Entering directory '/tmp/nix-build-perl-version-0.9912.drv-0/version-0.9912/vutil'
Running Mkbootstrap for version::vxs ()
chmod 644 "vxs.bs"
make[1]: Leaving directory '/tmp/nix-build-perl-version-0.9912.drv-0/version-0.9912/vutil'
PERL_DL_NONLAZY=1 "/nix/store/xlw7glm144q9k26r3v351i31z2rp0b99-perl-5.24.1/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/00impl-pp.t ..... ok

#   Failed test '"$version" eq "v1.2.3_4"'
#   at t/coretests.pm line 351.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'

#   Failed test '"$version" eq "v1.2.3_4" (from eval)'
#   at t/coretests.pm line 353.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'

#   Failed test '"$version" eq "v1.2.3_4"'
#   at t/coretests.pm line 351.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'

#   Failed test '"$version" eq "v1.2.3_4" (from eval)'
#   at t/coretests.pm line 353.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'

#   Failed test '"$version" eq "v1.2.3_4"'
#   at t/coretests.pm line 351.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'

#   Failed test '"$version" eq "v1.2.3_4" (from eval)'
#   at t/coretests.pm line 353.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'

#   Failed test '"$version" eq "v1.2.3_4"'
#   at t/coretests.pm line 351.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'

#   Failed test '"$version" eq "v1.2.3_4" (from eval)'
#   at t/coretests.pm line 353.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'
# Looks like you failed 8 tests of 646.
t/01base.t ........
Dubious, test returned 8 (wstat 2048, 0x800)
Failed 8/646 subtests
  2 3253k    2 77230    0     0  27265      0  0:02:02  0:00:02  0:02:00 78725
#   Failed test '"$version" eq "v1.2.3_4"'
#   at t/coretests.pm line 351.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'

#   Failed test '"$version" eq "v1.2.3_4" (from eval)'
#   at t/coretests.pm line 353.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'

#   Failed test '"$version" eq "v1.2.3_4"'
#   at t/coretests.pm line 351.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'

#   Failed test '"$version" eq "v1.2.3_4" (from eval)'
#   at t/coretests.pm line 353.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'

#   Failed test '"$version" eq "v1.2.3_4"'
#   at t/coretests.pm line 351.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'

#   Failed test '"$version" eq "v1.2.3_4" (from eval)'
#   at t/coretests.pm line 353.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'

#   Failed test '"$version" eq "v1.2.3_4"'
#   at t/coretests.pm line 351.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'

#   Failed test '"$version" eq "v1.2.3_4" (from eval)'
#   at t/coretests.pm line 353.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'
# Looks like you failed 8 tests of 657.
t/02derived.t .....
Dubious, test returned 8 (wstat 2048, 0x800)
Failed 8/657 subtests

#   Failed test '"$version" eq "v1.2.3_4"'
#   at t/coretests.pm line 351.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'

#   Failed test '"$version" eq "v1.2.3_4" (from eval)'
#   at t/coretests.pm line 353.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'

#   Failed test '"$version" eq "v1.2.3_4"'
#   at t/coretests.pm line 351.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'

#   Failed test '"$version" eq "v1.2.3_4" (from eval)'
#   at t/coretests.pm line 353.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'
# Looks like you failed 4 tests of 324.
t/03require.t .....
Dubious, test returned 4 (wstat 1024, 0x400)
Failed 4/324 subtests
        (less 18 skipped subtests: 302 okay)
t/04strict_lax.t .. ok
t/05sigdie.t ...... ok
t/06noop.t ........ ok
t/07locale.t ...... ok
t/08_corelist.t ... ok
t/09_list_util.t .. ok

Test Summary Report
-------------------
t/01base.t      (Wstat: 2048 Tests: 646 Failed: 8)
  Failed tests:  95-96, 255-256, 415-416, 575-576
  Non-zero exit status: 8
t/02derived.t   (Wstat: 2048 Tests: 657 Failed: 8)
  Failed tests:  101-102, 262-263, 423-424, 584-585
  Non-zero exit status: 8
t/03require.t   (Wstat: 1024 Tests: 324 Failed: 4)
  Failed tests:  98-99, 258-259
  Non-zero exit status: 4
Files=10, Tests=2466,  2 wallclock secs ( 0.33 usr  0.03 sys +  1.23 cusr  0.10 csys =  1.69 CPU)
Result: FAIL
Failed 3/10 test programs. 20/2466 subtests failed.
#   at t/coretests.pm line 353.
#     expected: 'v1.2.3_4'
# Looks like you failed 8 tests of 657.
t/02derived.t .....
Dubious, test returned 8 (wstat 2048, 0x800)
Failed 8/657 subtests

#   Failed test '"$version" eq "v1.2.3_4"'
#   at t/coretests.pm line 351.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'

#   Failed test '"$version" eq "v1.2.3_4" (from eval)'
#   at t/coretests.pm line 353.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'

#   Failed test '"$version" eq "v1.2.3_4"'
#   at t/coretests.pm line 351.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'

#   Failed test '"$version" eq "v1.2.3_4" (from eval)'
#   at t/coretests.pm line 353.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'
# Looks like you failed 4 tests of 324.
t/03require.t .....
Dubious, test returned 4 (wstat 1024, 0x400)
Failed 4/324 subtests
        (less 18 skipped subtests: 302 okay)
t/04strict_lax.t .. ok
t/05sigdie.t ...... ok
t/06noop.t ........ ok
t/07locale.t ...... ok
t/08_corelist.t ... ok
t/09_list_util.t .. ok

Test Summary Report
-------------------
t/01base.t      (Wstat: 2048 Tests: 646 Failed: 8)
  Failed tests:  95-96, 255-256, 415-416, 575-576
  Non-zero exit status: 8
t/02derived.t   (Wstat: 2048 Tests: 657 Failed: 8)
  Failed tests:  101-102, 262-263, 423-424, 584-585
  Non-zero exit status: 8
t/03require.t   (Wstat: 1024 Tests: 324 Failed: 4)
  Failed tests:  98-99, 258-259
  Non-zero exit status: 4
#     expected: 'v1.2.3_4'
# Looks like you failed 8 tests of 657.
t/02derived.t .....
Dubious, test returned 8 (wstat 2048, 0x800)
Failed 8/657 subtests

#   Failed test '"$version" eq "v1.2.3_4"'
#   at t/coretests.pm line 351.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'

#   Failed test '"$version" eq "v1.2.3_4" (from eval)'
#   at t/coretests.pm line 353.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'

#   Failed test '"$version" eq "v1.2.3_4"'
#   at t/coretests.pm line 351.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'

#   Failed test '"$version" eq "v1.2.3_4" (from eval)'
#   at t/coretests.pm line 353.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'
# Looks like you failed 4 tests of 324.
t/03require.t .....
Dubious, test returned 4 (wstat 1024, 0x400)
Failed 4/324 subtests
        (less 18 skipped subtests: 302 okay)
t/04strict_lax.t .. ok
t/05sigdie.t ...... ok
t/06noop.t ........ ok
t/07locale.t ...... ok
t/08_corelist.t ... ok
t/09_list_util.t .. ok

Test Summary Report
-------------------
t/01base.t      (Wstat: 2048 Tests: 646 Failed: 8)
  Failed tests:  95-96, 255-256, 415-416, 575-576
  Non-zero exit status: 8
t/02derived.t   (Wstat: 2048 Tests: 657 Failed: 8)
  Failed tests:  101-102, 262-263, 423-424, 584-585
  Non-zero exit status: 8
t/03require.t   (Wstat: 1024 Tests: 324 Failed: 4)
  Failed tests:  98-99, 258-259
  Non-zero exit status: 4
#     expected: 'v1.2.3_4'
# Looks like you failed 8 tests of 657.
t/02derived.t .....
Dubious, test returned 8 (wstat 2048, 0x800)
Failed 8/657 subtests

#   Failed test '"$version" eq "v1.2.3_4"'
#   at t/coretests.pm line 351.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'

#   Failed test '"$version" eq "v1.2.3_4" (from eval)'
#   at t/coretests.pm line 353.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'

#   Failed test '"$version" eq "v1.2.3_4"'
#   at t/coretests.pm line 351.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'

#   Failed test '"$version" eq "v1.2.3_4" (from eval)'
#   at t/coretests.pm line 353.
#          got: 'v1.2.3_4'
#     expected: 'v1.2.3_4'
# Looks like you failed 4 tests of 324.
t/03require.t .....
Dubious, test returned 4 (wstat 1024, 0x400)
Failed 4/324 subtests
        (less 18 skipped subtests: 302 okay)
t/04strict_lax.t .. ok
t/05sigdie.t ...... ok
t/06noop.t ........ ok
t/07locale.t ...... ok
t/08_corelist.t ... ok
t/09_list_util.t .. ok

Test Summary Report
-------------------
t/01base.t      (Wstat: 2048 Tests: 646 Failed: 8)
  Failed tests:  95-96, 255-256, 415-416, 575-576
  Non-zero exit status: 8
t/02derived.t   (Wstat: 2048 Tests: 657 Failed: 8)
  Failed tests:  101-102, 262-263, 423-424, 584-585
  Non-zero exit status: 8
t/03require.t   (Wstat: 1024 Tests: 324 Failed: 4)
  Failed tests:  98-99, 258-259
  Non-zero exit status: 4
Files=10, Tests=2466,  2 wallclock secs ( 0.33 usr  0.03 sys +  1.23 cusr  0.10 csys =  1.69 CPU)
Result: FAIL
Failed 3/10 test programs. 20/2466 subtests failed.
make: *** [Makefile:944: test_dynamic] Error 255
builder for ‘/nix/store/h2bpcdyy2nfwanv988w5y9w94pc9l6q6-perl-version-0.9912.drv’ failed with exit code 2
cannot build derivation ‘/nix/store/957md29lqbn5zqs4pkczzkhiq7pppbgk-perl-Module-Metadata-1.000027.drv’: 1 dependencies couldn't be built
building path(s) ‘/nix/store/alrszd2y137bpf2ymgf6fl8sc2nc2pzr-ssh-askpass-wrapper’
cannot build derivation ‘/nix/store/ff7bffcc2ix7qxv7mi5vs69dh4j55pgy-perl-Test-Warnings-0.026.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/ckxc1svgm8yns3vmsl98hy1i4vrqmqz5-perl-DateTime-1.21.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/wwzxp49gzk7gz677xlc7i36w4cdnkhfs-perl-Finance-Quote-1.38.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/rmg9l876ff4fz5zbfwr3ss13nmikirrk-gnucash-2.4.15.drv’: 1 dependencies couldn't be built
building path(s) ‘/nix/store/f7055955qcax8ifa9djgrxq1swl2sn5d-sshd_config’
cannot build derivation ‘/nix/store/vl85xb3llsd1xk2akiiww3z3iiqz2q6p-system-path.drv’: 1 dependencies couldn't be built
```
### Steps to reproduce
Not really sure which packages calls this.

## Technical details

* System: (NixOS: `nixos-version`, Ubuntu/Fedora: `lsb_release -a`, ...)
17.09pre105622.f0fac3b578 (Hummingbird)

* Nix version: (run `nix-env --version`)
nix-env (Nix) 1.11.8

* Nixpkgs version: (run `nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion`)
"17.09pre106117.c90998d5cf"

**Comment by jerith666** [comment]
I bisected this down to being caused by f78c032 "perl: init 5.24.1 and set as default perl", so CC'ing @plumps & @Mic92.

**Comment by vcunat** [comment]
Fixed by 9ff08d24efc.  (I verified it builds now.)
